server/cache.js

```diff
@@ -30,117 +34,63 @@ middlewareRouter.use(async (req, res) => {
-    return err
-  })
-
-  const preventCacheReason = noCache ? 'redirect_detected' : null
-  return purgeCache({
-    url: newPath,
-    modified,
-    editEmail: preventCacheReason,
-    ignore: ['redirect', 'missing', 'modified']
-  }).catch((err) => {
-    if (err && err.message !== 'Not found') log.warn(`Failed purging redirect destination ${newPath}`, err)
-    throw err
-  })
+  return cache.set(id, {content, modified: newModified, id})
 }
 
 // expose the purgeCache method externally so that list can call while building tree
 exports.purge = purgeCache
 
-async function purgeCache({url, modified, editEmail, ignore}) {
+async function purgeCache({id, modified, editEmail, ignore}) {
   modified = modified || moment().subtract(1, 'hour').utc().format('YYYY-MM-DDTHH:mm:ss.SSS[Z]')
 
   const overrides = ignore && Array.isArray(ignore) ? new Set(ignore) : new Set().add(ignore)
   const shouldIgnore = (type) => overrides.has(type) || overrides.has('all') || overrides.has('1')
 
-  if (!url) throw new Error(`Can't purge cache without url! Given url was ${url}`)
+  if (!id) throw new Error(`Can't purge cache without a document id! Given id was ${id}`)
 
-  const data = await cache.get(url)
-  // compare current cache entry data vs this request
-  const {redirectUrl, noCache, html, modified: oldModified, purgeId: lastPurgeId} = data || {}
+  const data = await cache.get(id)
 
-  if (redirectUrl && !shouldIgnore('redirect')) throw new Error('Unauthorized')
-  // edit is considered its own override for everything but redirect
+  if (!data) return // if no currently cached item, don't attempt to purge
+
+  // compare current cache entry data vs this request
+  const {noCache, content, purgeId: lastPurgeId} = data || {}
+  const html = content && content.html
 
   // FIXME: this should be more robust
   if (editEmail && editEmail.includes('@')) {
-    log.info(`CACHE PURGE PERSIST for ${noCacheDelay}s (${editEmail}): ${url}`)
-    return cache.set(url, {noCache: true}, {ttl: noCacheDelay})
+    log.info(`CACHE PURGE PERSIST for ${noCacheDelay}s (${editEmail}): ${id}`)
+    return cache.set(id, {noCache: true}, {ttl: noCacheDelay})
   }
 
   const purgeId = `${modified}-${editEmail || ''}-${ignore}`
 
-  // if attempting to purge /trash but nothing has changed, skip.
-  if (purgeId === lastPurgeId && url === '/trash') return
-
-  // const isTrashed = url.split('/')[1] === 'trash'
   // try and dedupe extra requests from multiple pods (tidier logs)
-  if (purgeId === lastPurgeId && !shouldIgnore('all')) throw new Error(`Same purge id as previous request ${purgeId} for ${url}`)
+  if (purgeId === lastPurgeId && !shouldIgnore('all')) throw new Error(`Same purge id as previous request ${purgeId} for docId ${id}`)
+
   // by default, don't try to purge empty
-  if (!html && !shouldIgnore('missing')) throw new Error('Not found')
+  else if (!html && !shouldIgnore('missing')) throw new Error('Not found')
```

Comment:
Would be nice to omit the `else` here unless it's necessary, but would it make sense to remove some of these conditions as part of the cache refactor? They might no longer be needed...